### PR TITLE
Fixed and modularised butt code

### LINKED ISSUE #6873: HOLY FUCKING SHIT
In-Game report from asskek:
Server info: 26937 (Hippie Station)

>General description of the issue
you can no longer take things out of your ass

>What you expected to happen
you can no longer take things out of your ass

>What actually happened
you can no longer take things out of your ass

>Steps to reproduce if possible
you can no longer take things out of your ass

### PR DESCRIPTION
**Pull request by JohnGinnane**
The original buttcode affected some tg files and the inspect check was removed in the bulk upstream merge I did a few weeks ago. I have modularised the code somewhat so hopefully this won't happen again

:cl: JohnGinnane
fix: Fixed not being able to inspect butts
/:cl:

Fixes #6873

### CODE REVIEW

**Comment by JohnGinnane** [comment]
@tortellinitony for info

File: hippiestation/code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/mob/living/carbon/human/grabbedby(mob/living/user, supress_message = 0)
+	if (checkbuttinspect(user))
```
Comment (Carbonhell) [inline hippiestation/code/modules/mob/living/carbon/human/human_defense.dm:2]: Why call this both here and in species proc

Comment (Carbonhell) [reply hippiestation/code/modules/mob/living/carbon/human/human_defense.dm:2]: ?

Comment (JohnGinnane) [reply hippiestation/code/modules/mob/living/carbon/human/human_defense.dm:2]: It's removed from there

**Comment by JohnGinnane** [comment]
image

Thank you for reminding me

**Comment by tortellinitony** [comment]
Will it display text like "A grabs B" or play a grab sound or something?

When you are trying to insert an item, will it do an attack?

**Comment by JohnGinnane** [comment]
It's still the same butt code I just moved the parts that affect tg files to our own, so all your original code is still running

**Review by Carbonhell — APPROVED** [review]

**Pull request merged**